homeassistant/components/matter/update.py

```diff
@@ -0,0 +1,221 @@
+"""Matter update."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Any
+
+from chip.clusters import Objects as clusters
+from matter_server.common.errors import UpdateCheckError, UpdateError
+from matter_server.common.models import MatterSoftwareVersion
+
+from homeassistant.components.update import (
+    ATTR_LATEST_VERSION,
+    UpdateDeviceClass,
+    UpdateEntity,
+    UpdateEntityDescription,
+    UpdateEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.restore_state import ExtraStoredData
+
+from .entity import MatterEntity
+from .helpers import get_matter
+from .models import MatterDiscoverySchema
+
+SCAN_INTERVAL = timedelta(hours=12)
```

Comment:
I'm surprised by this combined with `should_poll` not being overridden in the update entity class.
Is the `matter` integration polled?

Reply:
No, the Matter integration is not polled.

But checking for updates is. The device don't have a bulit-in update checking mechanism, so we don't get a notification pushed when there is an update.

We *could* do the polling on Matter Server side. But it feels more naturally to use Core's polling capability. With this, one can also use the `homeassistant.update_entity` service to force an update check.